pinot-common/src/main/java/org/apache/pinot/pql/parsers/PinotQuery2BrokerRequestConverter.java

```diff
@@ -0,0 +1,235 @@
+
+public class PinotQuery2BrokerRequestConverter {
+
+  public BrokerRequest convert(PinotQuery pinotQuery) {
+    BrokerRequest brokerRequest = new BrokerRequest();
+    Selection selection = null;
+    List<AggregationInfo> aggregationInfoList = null;
+    GroupBy groupBy = null;
+
+    //Query Source
+    QuerySource querySource = new QuerySource();
+    querySource.setTableName(pinotQuery.getDataSource().getTableName());
+    brokerRequest.setQuerySource(querySource);
+
+    Expression filterExpression = pinotQuery.getFilterExpression();
+
+    //Handle filter
+    if (filterExpression != null) {
+      FilterQuery filterQuery;
+      FilterQueryMap filterSubQueryMap = new FilterQueryMap();
+      filterQuery = traverseFilterExpression(filterExpression, filterSubQueryMap);
+      brokerRequest.setFilterQuery(filterQuery);
+      brokerRequest.setFilterSubQueryMap(filterSubQueryMap);
+    }
+
+    //Handle select list
+
+    for (Expression expression : pinotQuery.getSelectList()) {
+      ExpressionType type = expression.getType();
+      switch (type) {
+        case LITERAL:
+          if (selection == null) {
+            selection = new Selection();
+          }
+          selection.addToSelectionColumns(expression.getLiteral().getValue());
+          break;
+        case IDENTIFIER:
+          if (selection == null) {
+            selection = new Selection();
+          }
+          selection.addToSelectionColumns(expression.getIdentifier().getName());
+          break;
+        case FUNCTION:
+          AggregationInfo aggInfo = buildAggregationInfo(expression.getFunctionCall());
+          if (aggregationInfoList == null) {
+            aggregationInfoList = new ArrayList<>();
+          }
+          aggregationInfoList.add(aggInfo);
+          break;
+      }
+    }
+
+    if (selection != null) {
+      selection.setOffset(pinotQuery.getOffset());
+      selection.setSize(pinotQuery.getLimit());
+      brokerRequest.setSelections(selection);
+    }
+
+    if (aggregationInfoList != null && aggregationInfoList.size() > 0) {
+      brokerRequest.setAggregationsInfo(aggregationInfoList);
+    }
+
+    //Handle group by
+    List<Expression> groupByList = pinotQuery.getGroupByList();
+    if (groupByList != null && groupByList.size() > 0) {
+
+      groupBy = new GroupBy();
+      for (Expression expression : groupByList) {
+        String expressionStr = standardizeExpression(expression);
+        groupBy.addToExpressions(expressionStr);
+      }
+      groupBy.setTopN(pinotQuery.getLimit());
+    }
+
+    //Query Type
+    QueryType queryType = new QueryType();
+    if (aggregationInfoList != null && aggregationInfoList.size() > 0) {
+      if (groupBy != null) {
+        queryType.setHasGroup_by(true);
+      } else {
+        queryType.setHasAggregation(true);
+      }
+    } else {
+      queryType.setHasSelection(true);
+    }
+    brokerRequest.setQueryType(queryType);
+
+    //TODO: these should not be part of the query?
```

Comment:
Or may be just rename PinotQuery to PinotQueryRequest and then they could be part of the request.

Reply:
I thought we will keep Pinot Query and BrokerRequest will contain PinotQuery.

Reply:
👍